highlight/src/c_lib.rs

```diff
@@ -145,42 +152,60 @@ pub extern "C" fn ts_highlight_buffer_new() -> *mut TSHighlightBuffer {
     }))
 }
 
+/// Deleteis a [`TSHighlighter`] instance.
+///
+/// # Safety
+///
+/// `this` must be non-null.
 #[no_mangle]
-pub extern "C" fn ts_highlighter_delete(this: *mut TSHighlighter) {
-    drop(unsafe { Box::from_raw(this) })
+pub unsafe extern "C" fn ts_highlighter_delete(this: *mut TSHighlighter) {
+    drop(Box::from_raw(this))
 }
 
+/// Deleteis a [`TSHighlightBuffer`] instance.
+///
+/// # Safety
+///
+/// `this` must be non-null.
```

Comment:
If you don't have a tension to write a real docs for all this C API funcs right now I'd suggest to remove all such template doc messages because they don't provide useful meaning.

Every function that receives a raw pointer is unsafe by default.